eis_toolkit/transformations/binarize.py

```diff
@@ -0,0 +1,101 @@
+import numpy as np
+import rasterio
+
+from numbers import Number
+from beartype import beartype
+from beartype.typing import Optional, Tuple, Sequence
+
+from eis_toolkit.utilities.miscellaneous import expand_and_zip, cast_scalar_to_int, check_dtype_for_int
+
+from eis_toolkit.exceptions import InvalidRasterBandException, NonMatchingParameterLengthsException
+from eis_toolkit.checks.raster import check_raster_bands
+from eis_toolkit.checks.parameter import check_parameter_length
+
+
+@beartype
+def _binarize(
+    in_array: np.ndarray,
+    threshold: Number,
+) -> np.ndarray:  # type: ignore[no-any-unimported]
+    out_array = np.where(in_array <= threshold, 0, 1)
+
+    return out_array
+
+
+@beartype
+def binarize(  # type: ignore[no-any-unimported]
+    raster: rasterio.io.DatasetReader,
+    bands: Optional[Sequence[int]] = None,
+    thresholds: Sequence[Number] = [Number],
```

Comment:
Maybe leave the default parameter out here too? Anyway, I believe `[Number]` is not a valid default parameter

Reply:
Hi, I don't see a notebook included in the PR right now, or do you mean you tested it just locally? Anyway, when I modified the test files so that I removed `limits=limits` or `percentiles=percentiles` from the function calls (making the functions use the default values), the tests failed raising exceptions `InvalidParameterValueException: Limit values all None: (None, None).` or `InvalidParameterValueException: Percentile values all None: (None, None)`.

I think the separation makes sense if it is often the case that the user wants to give one limit/percentile but leave the other as None. Then I would go for your approach 1. Otherwise if we want to define these inputs as a tuple, I would just remove the default values and force the user to always define inputs, for example `limits = (5, None)`.

We can keep possibility to process multiple bands since it has been implemented already. Better to change it only when we really see a need for it and not discard work already done :+1:.